doc/source/cluster/kubernetes/k8s-ecosystem/yunikorn.md

```diff
@@ -0,0 +1,158 @@
+
+## Setup
+
+### Step 1: Create a Kubernetes cluster with KinD
+Run the following command in a terminal:
+
+```shell
+kind create cluster
+```
+
+### Step 2: Install Apache YuniKorn
+
+You need to successfully install Apache YuniKorn on your Kubernetes cluster before enabling Apache YuniKorn integration with KubeRay.
+See [Get Started](https://yunikorn.apache.org/docs/) for Apache YuniKorn installation instructions.
+
+### Step 3: Install the KubeRay Operator with Apache YuniKorn support
+
+Deploy the KubeRay Operator with the `--enable-batch-scheduler` flag to enable Volcano batch scheduling support.
+
+When installing KubeRay Operator using Helm, you should use one of these two options:
+
+* Set `batchScheduler.name` to `yunikorn` in your
+[`values.yaml`](https://github.com/ray-project/kuberay/blob/df5577fe1b3f537f36dbda3870b4743edd2f11bb/helm-chart/kuberay-operator/values.yaml#L83)
+file:
+```shell
+# values.yaml file
+batchScheduler:
+    name: yunikorn
+```
+
+* Pass the `--set batchScheduler.name=yunikorn` flag when running on the command line:
+```shell
+# Install the Helm chart with --enable-batch-scheduler flag set to true
+helm install kuberay-operator kuberay/kuberay-operator --namespace ray-system --version 1.2.2 --create-namespace --set batchScheduler.name=yunikorn
+```
+
+### Step 4: Use Apache YuniKorn for batch scheduling
+
+## Example
+
+### Gang scheduling
+
+This example walks through how gang scheduling works with Apache YuniKorn and KubeRay.
+
+First, create a queue with a capacity of 4 CPUs and 6Gi of RAM by editing the ConfigMap:
+
+Run `kubectl edit configmap -n yunikorn yunikorn-defaults`
+
+Add a `queues.yaml` config to under the `data` key, the final ConfigMap should look like this:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  # Metadata for the ConfigMap, skip for brevity
+data:
+  queues.yaml: |
+    partitions:
+      - name: default
+        queues:
+          - name: root
+            queues:
+              - name: test
+                submitacl: "*"
+                parent: false
+                resources:
+                  guaranteed:
+                    memory: 6G
+                    vcore: 4
+                  max:
+                    memory: 6G
+                    vcore: 4
+```
+
+Save the changes and exit the editor. This configuration creates a queue named `root.default` with a capacity of 4 CPUs and 6Gi of RAM.
+
+Next, create a RayCluster with a head node (1 CPU + 2Gi of RAM) and two workers (1 CPU + 1Gi of RAM each), for a total of 3 CPU and 4Gi of RAM:
+
+
+```shell
+# Path: kuberay/ray-operator/config/samples
+# Includes the necessary labels for the Apache YuniKorn scheduler gang scheduling:
+# - `ray.io/gang-scheduling-enabled`
+# - `yunikorn.apache.org/app-id`
+# - `yunikorn.apache.org/queue`
+wget https://raw.githubusercontent.com/ray-project/kuberay/v1.2.2/ray-operator/config/samples/ray-cluster.yunikorn-scheduler.yaml
+kubectl apply -f ray-cluster.yunikorn-scheduler.yaml
+```
```

Comment:
Could we briefly explain these three labels so that users understand how to configure them? For example, it's not clear why `yunikorn.apache.org/queue` should be set to `root.test`.

```yaml
labels:
  ray.io/gang-scheduling-enabled: "true"
  yunikorn.apache.org/app-id: test-yunikorn-0
  yunikorn.apache.org/queue: root.test
```